### PR TITLE
CPM-899: Add validation for delimiter

### PR DESCRIPTION
**Pull request by cemorin**
**Description (for Contributor and Core Developer)**

Delimiter can now be anything but spaces, comma or semicolon

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by Balamung — APPROVED** [review]

File: components/identifier-generator/back/tests/features/updateIdentifierGenerator.feature [hunk 1]
```diff
@@ -125,3 +125,15 @@ Feature: Update Identifier Generator
   Scenario: Cannot update an identifier generator with delimiter too long
     When I try to update an identifier generator with delimiter 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec suscipit nisi erat, sed tincidunt urna finibus non. Nullam id lacus et augue ullamcorper euismod sed id nibh. Praesent luctus cursus finibus. Maecenas et euismod tellus. Nunc sed est nec mi consequat consequat sit amet ac ex. '
     Then I should get an error on update with message 'delimiter: This value is too long. It should have 100 characters or less.'
```
Comment (pierallard) [inline components/identifier-generator/back/tests/features/updateIdentifierGenerator.feature:127]: Can you add at least one test in createIdentifierGenerator too ?

**Review by pierallard — APPROVED** [review]

**Review by Beninho — APPROVED** [review]

File: components/identifier-generator/back/tests/features/updateIdentifierGenerator.feature [hunk 2]
```diff
@@ -63,6 +63,18 @@ Feature: Update Identifier Generator
     When I try to update an identifier generator with free text with unknown field
     Then I should get an error on update with message 'structure[0][unknown]: This field was not expected.'
 
+  Scenario: Cannot update and identifier generator if structure contains free text with invalid text
+    When I try to update an identifier generator with free text ' AKN'
+    Then I should get an error on update with message 'Free text must not contain a comma, a semicolon or any space'
+
+  Scenario: Cannot update and identifier generator if structure contains free text with invalid text
+    When I try to update an identifier generator with free text ','
```
Comment (Beninho) [inline components/identifier-generator/back/tests/features/updateIdentifierGenerator.feature:71]: Maybe test with words separated by comma, like 'product,color' or something

File: components/identifier-generator/back/tests/features/updateIdentifierGenerator.feature [hunk 3]
```diff
@@ -63,6 +63,18 @@ Feature: Update Identifier Generator
...
+    Then I should get an error on update with message 'Free text must not contain a comma, a semicolon or any space'
+
+  Scenario: Cannot update and identifier generator if structure contains free text with invalid text
+    When I try to update an identifier generator with free text ';'
```
Comment (Beninho) [inline components/identifier-generator/back/tests/features/updateIdentifierGenerator.feature:75]: Same comment with test ';'

File: components/identifier-generator/back/tests/features/updateIdentifierGenerator.feature (hunk 2 above)
Comment (cemorin) [reply components/identifier-generator/back/tests/features/updateIdentifierGenerator.feature]: Yep, that's a better test :) done :+1:

**Review by pierallard — APPROVED** [review]

**Pull request closed without merging**